packages/lib-runtime/src/schema/plugin-manifest.ts

```diff
@@ -45,21 +45,24 @@ const extensionType = yup
  * Schema for `Extension` objects.
  */
 const extension = yup
-  .object({
+  .object()
+  .required()
+  .shape({
     type: extensionType,
     properties: yup.object().required(),
   })
-  .required();
+  .noUnknown(false);
```

Comment:
This reflects `[customProperty: string]: unknown;` properties in `Extension` type, allowing extension declarations to contain custom (direct) properties.

Reply:
`yup` doesn't lock down extra properties. It validates the structure to match and ignores unknown properties by default. We don't need `noUnknown` as `required` covers the effort of additional properties while enforcing the object is there (you don't need the earlier required in this case)

Reply:
Quick clarity of `yup` usage:

`.object()` is needed to do `.shape`
`.required()` is desired any point we want this just for clarity; but honestly the fact that props internal are required, it makes the object required
`.shape({ ... })` is needed from a "we want structure" point of view and thus also makes it required if internal properties are required